File: pset6_2018/cash/cash.py
from cs50 import get_float


def main():
    change = prompt()
    amount = find_min(round(change * 100))
    print(amount)


def prompt():
    change = get_float('Change owed: ')
    if change < 0:
        return prompt()
    return change


def find_min(cents):
    coins = [25, 10, 5]
    minimum = 0
    for x in range(3):
        minimum += int(cents / coins[x])
        cents %= coins[x]
    return minimum + cents


if __name__ == '__main__':
    main()

File: pset1_2017/cash.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <cs50.h>

int main(void)
{
    float change = 0;
    do
    {
        printf("Change owed: ");
        change = get_float();
    }
    while (change < 0);

    // array with with values for each coin (in cents)
    int cents = round(change * 100);
    int coins[3] = {25, 10, 5};
    int minimum = 0;

    // determining minimum number of coins
    for (int i = 0; i < 3; i++)
    {
        minimum += cents / coins[i];
        cents %= coins[i];
    }

    printf("%i\n", minimum + cents);
    return 0;
}

